resources/puppetlabs/lein-ezbake/template/global/ext/fpm.rb

```diff
@@ -189,6 +189,33 @@
     options.java = 'java-11-openjdk-headless'
   end
 
+  # patch the sysconfig file so it has the correct JAVA_BIN
+  default_paths = [
+    "#{options.chdir}/etc/sysconfig/puppet*", # EL family
+    "#{options.chdir}/etc/default/puppet*",   # Debian family
+  ]
+  target = "JAVA_BIN=#{options.java_bin}"
+  Dir.glob(default_paths).each do |file_path|
+    lines = File.readlines(file_path).map do |line|
+      line.start_with?('JAVA_BIN=') ? target : line
+    end
+    File.write(file_path, lines.join)
+    puts "patched JAVA_BIN in #{file_path} to #{target}"
+  end
+
+  # patch the systemd unit file to have the correct JAVA_BIN
+  systemd_paths = [
+    "#{options.chdir}/usr/lib/systemd/system/puppet*.service", # EL family
+    "#{options.chdir}/lib/systemd/system/puppet*.service",     # Debian family
+  ]
+  searchstring = '/usr/bin/java'
```

Comment:
How come we take `options.java_bin` above and hard code it here?

Reply:
Seems we don't use this path in https://github.com/OpenVoxProject/ezbake/blob/cfe369e88f66a06ef6475acd52f609edbebde09c/resources/puppetlabs/lein-ezbake/template/global/ext/fpm.rb#L170-L178, but maybe /usr/bin/java is just a symlink?